templates/windows/chocolateyInstall.ps1.erb

```diff
@@ -1,26 +1,40 @@
 $toolsDir = "$(Split-Path -parent $MyInvocation.MyCommand.Definition)"
+$fileList = Join-Path -path $toolsDir -childpath "file-list.txt"
 
-<%- dirnames = get_directories.map {|d| d.path } -%>
-<%- dirnames.each do |dir| -%>
-  <%- windows_path = platform.convert_to_windows_path(dir) -%>
-if (Test-Path -path "<%= windows_path %>") {
-  Write-Debug "Warning: copying over '<%= windows_path %>'"
-} else {
-  New-Item -ItemType directory -Path "<%= windows_path %>"
+if (!(Test-Path -path "$fileList")) {
+  Write-Host "Unable to find file '$fileList', cannot proceed with install"
+  exit
 }
-$origin = Join-Path -path "$toolsDir" -childpath "<%= dir.gsub('/', '\\') %>"
-Copy-Item -path "$origin\*" -destination "<%= windows_path.gsub('/', '\\') %>\" -Recurse -Force
-<%- end -%>
 
-<%- filenames = get_files.map {|f| f.path} -%>
-<%- configfilenames = get_configfiles.map {|cf| cf.path} -%>
-<%- (filenames + configfilenames).each do |file| -%>
-  <%- windows_path = platform.convert_to_windows_path(file) -%>
-if (Test-Path -path "<%= windows_path %>") {
-  Write-Debug "Warning: copying over '<%= windows_path %>'"
-} elseif (!(Test-Path -path "<%= File.dirname windows_path %>")) {
-  New-Item -ItemType directory -Path "<%= File.dirname windows_path %>"
+$reader = [System.IO.File]::OpenText("$fileList")
+try {
+  for (;;) {
+    $destination = $reader.ReadLine()
+    if ($destination -eq $null) { break }
+    $originFile = "$destination" -replace '^[a-zA-Z]:/',''
+    $origin = Join-Path -path "$toolsDir" -childpath "$originFile"
+    if (Test-Path -path "$origin") {
+      if ((Get-Item "$origin") -is [System.IO.DirectoryInfo]) {
+        if (!( Test-Path -path "$destination")) {
+          New-Item -ItemType directory -Path "$destination"
+        }
+      } else {
+        $parent = Split-Path -path $destination -parent
+        if (!(Test-Path -path "$parent")) {
+          New-Item -ItemType directory -Path "$parent"
+        }
+        if (Test-Path -path "$destination") {
+          Write-Host "Overwriting '$destination'"
+        }
+        Copy-Item -path "$origin" -destination "$destination" -Force
+      }
+    } else {
+      if (!(Test-Path -path "$destination")) {
+        New-Item -ItemType directory -Path "$destination"
+      }
+    }
+  }
+}
+finally {
+  $reader.Close()
```

Comment:
None in particular. I forgot we were doing something similar in the uninstall script, and this was the first thing I found on the internet that seemed to work. Is there a better way of doing this?